Refactor: Use Angular inject() for Dependency Injection Across All Modules - Issue 23231

fahrigedik opened the pull request:
### Description

Resolves #23231 

This PR refactors all Angular services, components, and directives in the repository to use the Angular v14+ inject() function for dependency injection instead of constructor-based DI. All DI dependencies are now initialized as class properties using inject(), and constructors have been removed or simplified where appropriate. This change aligns the codebase with Angular's latest best practices for dependency injection, improves code readability, and prepares the project for future Angular versions.

Highlights:

All services, components, and directives now use inject() for DI.
Constructors that only existed for DI have been removed.
No functional or API changes were made; only the DI pattern was updated.
No breaking changes for consumers unless they relied on constructor injection in extended classes.
No screenshots or GIFs are required as this is a code refactor.

### Checklist

- [x] I fully tested it as developer / designer and created unit / integration tests
- [x] I documented it (or no need to document or I will create a separate documentation issue)

### How to test it?
Run the application and ensure all features work as before.
Pay special attention to areas where services, components, or directives are injected.
Run all existing unit and integration tests to confirm there are no regressions.


for @sumeyyeKurtulus 
### Migration Options & Rationale
1. Migration Path: [abp](vscode-file://vscode-app/c:/Users/fahri/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)
Choice: Migrate entire project
Rationale: Ensure consistency across all code and avoid mixed DI patterns.

2. Abstract Classes Migration: Yes
3. Backwards Compatible Constructors: **No**
4. Optional Inject Non-nullable: Yes

fahrigedik committed:
refactor: migrate account package to inject() DI

fahrigedik committed:
refactor : migrate components package to inject() DI

fahrigedik committed:
refactor : migrate core package to inject() DI

fahrigedik committed:
refactor : migrate feature-management package to inject() DI

fahrigedik committed:
refactor : migrate identity package to inject() DI

fahrigedik committed:
refactor : migrate oauth package to inject() DI

fahrigedik committed:
refactor : migrate permission-management package to inject() DI

fahrigedik committed:
refactor : migrate setting-management package to inject() DI

fahrigedik committed:
refactor : migrate tenant-management package to inject() DI

fahrigedik committed:
refactor : migrate theme-basic package to inject() DI

fahrigedik committed:
refactor : migrate theme-shared package to inject() DI

fahrigedik committed:
refactor : migrate schematics template to inject() DI

fahrigedik committed:
Remove unused service and variable injections

Eliminated unused injections of RestService, Router, and ViewContainerRef from MultiTenancyService, LoaderBarComponent, and LoadingDirective respectively to clean up the code and improve maintainability.

fahrigedik committed:
Refactor LocaleId to use inject for LocalizationService

Updated the LocaleId class to use Angular's inject() function for obtaining the LocalizationService instead of constructor injection. Removed the deps array from the LocaleProvider as it is no longer needed.

fahrigedik committed:
Remove unnecessary injector parameter from constructors

Updated PageToolbarComponent and LazyStyleHandler initialization to remove the unused Injector parameter from their constructors and factory functions. This simplifies instantiation and aligns with recent refactoring.

sumeyyeKurtulus commented on npm/ng-packs/packages/components/extensible/src/lib/components/grid-actions/grid-actions.component.ts:
```diff
@@ -43,7 +44,9 @@ export class GridActionsComponent<R = any> extends AbstractActionsComponent<Enti
-  constructor(injector: Injector) {
-    super(injector);
+  constructor() {
+    const injector = inject(Injector);
```
We can remove this `injector` declaration since it is not used.

sumeyyeKurtulus commented on npm/ng-packs/packages/schematics/src/commands/api/files-service/proxy/__namespace@dir__/__name@kebab__.service.ts.template:
```diff
@@ -5,6 +5,7 @@
   providedIn: 'root',
 })
 export class <%= name %>Service {
+  private restService = inject(RestService);
```
We need to add missing `inject` import here.

sumeyyeKurtulus commented on npm/ng-packs/packages/theme-shared/src/lib/directives/ellipsis.directive.ts:
```diff
@@ -1,17 +1,21 @@
+  ElementRef, 
+  HostBinding, 
+  Input, 
+  NgModule, 
```
This import is not used. We can remove that.

sumeyyeKurtulus commented on npm/ng-packs/packages/theme-shared/src/lib/components/loader-bar/loader-bar.component.ts:
```diff
@@ -1,4 +1,4 @@
-import { ChangeDetectorRef, Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { ChangeDetectorRef, Component, Input, OnDestroy, OnInit, inject } from '@angular/core';
 import { Router } from '@angular/router';
```
This import is not used here either.

sumeyyeKurtulus commented on npm/ng-packs/packages/theme-shared/src/lib/directives/loading.directive.ts:
```diff
@@ -1,16 +1,17 @@
+  OnDestroy, 
+  OnInit, 
+  Renderer2, 
+  ViewContainerRef, 
```
`ViewContainerRef` import should be removed either.

sumeyyeKurtulus commented on npm/ng-packs/packages/theme-shared/src/lib/utils/date-parser-formatter.ts:
```diff
@@ -13,9 +13,9 @@ function toInteger(value: any): number {
-  constructor(private configState: ConfigStateService, @Inject(LOCALE_ID) private locale: string) {
-    super();
-  }
+  private configState = inject(ConfigStateService);
```
We can keep this super call here just in case. This is because we extend an external formatter called `NgbDateParserFormatter`

sumeyyeKurtulus reviewed (changes requested):
Hello @fahrigedik , it looks nice and neat other than the points I mentioned above.

fahrigedik committed:
Remove unused imports and minor constructor cleanup

Eliminated unused imports from several components and directives to improve code clarity and reduce bundle size. Added an explicit constructor to DateParserFormatter for consistency. No functional changes were made.

sumeyyeKurtulus committed:
update: remove unused imports from `GridActionsComponent`

yagmurcelk reviewed (changes requested):
I get an error when I run this command (`yarn nx build core --skip-nx-cache --verbose`)

![image](https://github.com/user-attachments/assets/0e2a309b-a56f-49d3-9e13-49d787a94b10)

fahrigedik committed:
Refactor mock services to simplify super constructor calls

Updated MockPermissionService and MockRestService to call their super constructors without arguments, aligning with possible changes in the base class constructors. This simplifies the instantiation logic and removes unnecessary dependencies from the mock implementations.

fahrigedik commented:
> I get an error when I run this command (`yarn nx build core --skip-nx-cache --verbose`)
> 
> ![image](https://private-user-images.githubusercontent.com/157795849/464135731-0e2a309b-a56f-49d3-9e13-49d787a94b10.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIwNTg1NjMsIm5iZiI6MTc1MjA1ODI2MywicGF0aCI6Ii8xNTc3OTU4NDkvNDY0MTM1NzMxLTBlMmEzMDliLWE1NmYtNDlkMy05ZTEzLTQ5ZDc4N2E5NGIxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcwOVQxMDUxMDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZGM5Njc5MzY4ODkzNzA1NzcxYTg2NDkwNWY1OWYwZmFiNGUzNmYzOWQ0MWM4Nzc4NWFkZDU3NmE1ZDE5ZjMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.0AX7GLfghNmzyUqnSg8iw1ADSjuuKQQQ2b8OHuro-ZI)

i fixed this error

yagmurcelk reviewed (changes requested):
I am getting an error on the volo side

![image](https://github.com/user-attachments/assets/8f79834e-4fd8-42cb-9c82-eff88a3ea3f5)

yagmurcelk reviewed (changes requested):
I get an error when I run the `yarn copy-to:app-pro` command on the volo side
<img width="1701" height="996" alt="image" src="https://github.com/user-attachments/assets/1cd4c46b-91c7-408e-b476-61b13762c8ad" />

yagmurcelk reviewed (changes requested):
- [ ] <img width="1871" height="1372" alt="image" src="https://github.com/user-attachments/assets/1e687573-20aa-46a6-aa61-925f3e859acd" />
- [ ] It gives an error in the console on the login page.
<img width="2554" height="1470" alt="image" src="https://github.com/user-attachments/assets/6e434ab9-1aab-4b8f-b2ad-93acf6046a4e" />

- [ ] After the user logs in, the pages do not work normally for a while, but then they are fixed. Even though I log in, the login button continues to appear.

[Apex_1752662699965.webm](https://github.com/user-attachments/assets/ad73cae1-b0ba-48b1-9c0f-497d1aceb6f3)


<img width="486" height="488" alt="image" src="https://github.com/user-attachments/assets/bdf4c68f-042d-4907-95fb-35048efdcd6b" />

fahrigedik committed:
Merge branch 'dev' into issue-23231-v2

fahrigedik committed:
Revert "Merge branch 'dev' into issue-23231-v2"

This reverts commit 7c2582fe3ffc9505d0b54464cda35dcf7015d4c1, reversing
changes made to 7621fc7fcdc584689ed8907edd4061e1aff5b9f3.

fahrigedik commented:
Hi @yagmurcelk These errors occur because the scripts are not loaded and I don't get them. would you like someone else to test it?

fahrigedik committed:
Merge branch 'dev' into issue-23231-v2

fahrigedik committed:
Reapply "Merge branch 'dev' into issue-23231-v2"

This reverts commit a83bce6b7b2617a121eb4305fb27bd342145d7ed.
